packages/server/src/auth/auth.service.ts

```diff
@@ -9,6 +9,22 @@ import { DatabaseService } from '../database/database.service';
 export class AuthService {
   constructor(private readonly database: DatabaseService) {}
 
+  /**
+   * Checks that a user/password pair exists in the database without verifying password
+   * @async
+   * @param {string} usernameValue
+   *
+   * @return {Promise<User|null>} promise - a promise with the validation results. If successful
+   * will return the user, otherwise it returns null.
+   */
+  async validateUserByEmail(email: string): Promise<User | null> {
+    const databaseUser: User = await this.database.users.findOne({
+      email,
+    });
+    if (!databaseUser || !databaseUser.enabled) return null;
+    return databaseUser;
```

Comment:
This might be nit-picky but generally I like to use an optimistic approach when writing conditional logic. Makes it easier to read:

```js
if (databaseUser?.enabled) return databaseUser;
return null;
```

Reply:
Interesting, I normally follow the pattern the other way around, more like validation style: First check whether data can be worked with, and then execute the logic. 

I took this pattern from Go, where you need to check for errors after any call that could lead to errors. In Go, you regularly end up with deeply nested if structures if you follow the approach you describe. The gocyclo linter will bite you then ;) 

Having said that, there is certainly no reason to adopt patterns from Go to JS, so I guess this really comes down to taste/preference.

Reply:
That's a good point. Yeah probably a separate discussion on code style. 👍